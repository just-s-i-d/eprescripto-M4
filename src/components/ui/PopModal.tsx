import { Modal } from "antd";
import { ReactNode, SetStateAction } from "react";

type ConfirmModalPropsType = {
  title: string;
  okButtonText?: string;
  open: boolean;
  setOpen: React.Dispatch<SetStateAction<boolean>>;
  children: ReactNode;
  footer?: boolean;
  className?: string;
  confirmHandler?: VoidFunction;
};
const PopModal = ({
  open,
  setOpen,
  children,
  title,
  okButtonText,
  footer,
  className,
  confirmHandler,
}: ConfirmModalPropsType) => {
  const handleOk = () => {
    if (confirmHandler) confirmHandler();
    setOpen(false);
  };

  return (
    <Modal
      className={className}
      open={open}
      title={title}
      centered
      okText={okButtonText || "Sure"}
      onOk={handleOk}
      onCancel={() => setOpen(false)}
      footer={
        footer
          ? (_, { OkBtn, CancelBtn }) => (
              <>
                <CancelBtn />
                <OkBtn />
              </>
            )
          : null
      }
    >
      {children}
    </Modal>
  );
};
export default PopModal;
