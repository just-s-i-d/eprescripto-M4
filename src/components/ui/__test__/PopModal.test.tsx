import { render, screen, fireEvent, waitFor } from "@testing-library/react";
import "@testing-library/jest-dom";
import PopModal from "../PopModal";

describe("PopModal Component", () => {
  it("Snapshot test", () => {
    const { asFragment } = render(
      <PopModal open={true} setOpen={jest.fn()} title="Test Title">
        <div>Test Content</div>
      </PopModal>,
    );
    expect(asFragment()).toMatchSnapshot();
  });

  it("renders PopModal with title and children", () => {
    render(
      <PopModal open={true} setOpen={jest.fn()} title="Test Title">
        <div>Test Content</div>
      </PopModal>,
    );
    expect(screen.getByText("Test Title")).toBeInTheDocument();
    expect(screen.getByText("Test Content")).toBeInTheDocument();
  });

  it("calls confirmHandler when OK button is clicked", async () => {
    const confirmHandler = jest.fn();
    render(
      <PopModal
        open={true}
        setOpen={jest.fn()}
        title="Test Title"
        confirmHandler={confirmHandler}
        okButtonText="Sure"
        footer={true}
      >
        <div>Test Content</div>
      </PopModal>,
    );
    fireEvent.click(screen.getByText("Sure"));
    await waitFor(() => {
      expect(confirmHandler).toHaveBeenCalled();
    });
  });

  it("closes the modal when Cancel button is clicked", () => {
    const setOpenMock = jest.fn();
    render(
      <PopModal
        open={true}
        setOpen={setOpenMock}
        title="Test Title"
        footer={true}
      >
        <div>Test Content</div>
      </PopModal>,
    );

    fireEvent.click(screen.getByText("Cancel"));
    expect(setOpenMock).toHaveBeenCalledWith(false);
  });
  it("closes the modal when Close button is clicked", () => {
    const setOpenMock = jest.fn();
    render(
      <PopModal
        open={true}
        setOpen={setOpenMock}
        title="Test Title"
        footer={true}
      >
        <div>Test Content</div>
      </PopModal>,
    );
    const closeBtn = screen.getByRole("button", { name: "Close" });
    fireEvent.click(closeBtn);
    expect(setOpenMock).toHaveBeenCalledWith(false);
  });
});
