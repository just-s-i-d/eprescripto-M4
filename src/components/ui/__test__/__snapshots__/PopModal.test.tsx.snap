// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`PopModal Component Snapshot test 1`] = `<DocumentFragment />`;
